### ISSUE TITLE
NPE when trying to render multi-valued widgets (List, Select, MultiSelect) with empty candidates expression

### ISSUE THREAD

**Issue by pcdavid** [eclipse-sirius/sirius-web#2072]
Newly created List/Select/MultiSelect widgets (in a View-based form) have empty expressions for theirs values/candidates.
This should mean "no candidate", but because of how `ViewFormDescriptionConverterSwitch.getMultiValueProvider(String)` is implemented this results in a non-null and non-empty list of candidates/items, with a single `""` (empty string) item.
This causes NPEs later when when e.g. trying to determine the id/label/kind of `""`.

Basically, an empty expression should mean "no candidates" (`List.of()`), and not as is the case now "a single candidate that is the empty string" (`List.of("")`).

It's quite annoying because it means these widgets are not "ready to use" (even in an empty but valid state) on creation, and one always has to provide explicit configuration for them.
